Core/src/ApiHelpersTrait.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Cloud\Core;
+
+/**
+ * @internal
+ * Supplies helper methods to interact with the APIs.
+ */
+trait ApiHelpersTrait
+{
+    /**
+     * Format a struct for the API.
+     *
+     * @param array $fields
+     * @return array
+     */
+    private function formatStructForApi(array $fields)
```

Comment:
These methods are not new and are currently present in `GrpcTrait`. This was used in the connection classes to pre-process/post-process data.

Since, we are removing the connection class, it seems fitting to extract it out and now it can be used by the clients/ resource classes(Topic, Subscription etc).